client/src/pages/public/Keycloak.js

```diff
@@ -0,0 +1,27 @@
+import React, { useEffect } from 'react';
+import { Redirect, } from 'react-router-dom';
+import { useKeycloak } from '@react-keycloak/web';
+import { Routes } from '../../constants';
+import store from 'store';
+
+// eslint-disable-next-line import/no-anonymous-default-export
+export default () => {
+  const [keycloak] = useKeycloak();
+
+  useEffect(() => {
+    if (keycloak.authenticated) {
+      store.set('TOKEN', keycloak.token);
+    }
+  }, [keycloak.authenticated, keycloak.token]);
+
+
+  return keycloak.authenticated
+    ?
+    <Redirect to={store.get('REDIRECT')} />
+    :
+    keycloak.loginRequired
+      ?
+      <Redirect to={Routes.Login} />
```

Comment:
I guess this actually takes care of the 401 page :)

Reply:
Although maybe not, I guess once we add user roles there will be a little more nuance where some screens might be inaccessible even to logged-in users, depending on their role

Reply:
The login screen will be centralized anyway, we can although pass parameters to the Login component in the future